### ISSUE TITLE
error: field has incomplete type 'XXH32_state_t' (aka 'XXH32_state_s')

### ISSUE THREAD

**Issue by tuananh** [mscdex/node-xxhash#18]
`npm i` on OS X returns these errors for me.

Any idea on how to fix this?

```> xxhash@0.2.4 install /Users/anh/code/node-xxhash
> node-gyp rebuild

  CXX(target) Release/obj.target/hash/src/hash.o
In file included from ../src/hash.cc:3:
../src/hash_32.hpp:12:19: error: field has incomplete type 'XXH32_state_t' (aka 'XXH32_state_s')
    XXH32_state_t state;
                  ^
../deps/xxhash/xxhash.h:170:16: note: forward declaration of 'XXH32_state_s'
typedef struct XXH32_state_s XXH32_state_t;   /* incomplete type */
               ^
In file included from ../src/hash.cc:3:
../src/hash_32.hpp:93:7: error: cannot initialize object parameter of type 'node::ObjectWrap' with an expression of type 'Hash32'
      obj->Wrap(info.This());
      ^~~
In file included from ../src/hash.cc:4:
../src/hash_64.hpp:12:19: error: field has incomplete type 'XXH64_state_t' (aka 'XXH64_state_s')
    XXH64_state_t state;
                  ^
../deps/xxhash/xxhash.h:228:16: note: forward declaration of 'XXH64_state_s'
typedef struct XXH64_state_s XXH64_state_t;   /* incomplete type */
               ^
In file included from ../src/hash.cc:4:
../src/hash_64.hpp:93:7: error: cannot initialize object parameter of type 'node::ObjectWrap' with an expression of type 'Hash64'
      obj->Wrap(info.This());
      ^~~
4 errors generated.```

**Comment by mscdex** [comment]
I'm working on updating upstream xxhash which should fix this and other issues.

**Comment by tuananh** [comment]
@mscdex ah i see. I'm working on that too and when remove `deps/xxhash` and re-add as submodule, i'm seeing this issue.

Cheers,

**Comment by mscdex** [comment]
This should be fixed in b9980c2.

**Comment by tuananh** [comment]
Cool. Thanks. Lemme try again.
